### PR TITLE
Modify firm model to include stringified CEO, CFO and COO

### PR DESCRIPTION
**Pull request by keanucz**
It has been bugging me for the past week now. Again, with MemeBot, I want to display the CEO, CFO and COO but this is not possible without doing an API call to fetch over 52 investor profiles JUST to single out the CEO, CFO or COO. This change to the firm model would reduce load and would also make it easier.

Although I have not tested this change yet, it is entirely feasible to adapt to the new model with the use of a migration script. Query the investors table, fetch people with CEO roles, assign the CEO value of their firm with their reddit username, and so on with CFO and COO. 

I could easily write an SQL query script that would do this, waiting for your feedback.

(Yes, I know you'll also have to modify the DB but it's not hard for such a good change. Only a few minutes of maintenance to just modify columns and then run the migration script)

### CODE REVIEW

**Comment by keanucz** [comment]
TODO: fix python codebase to not increment CFO/coo value by 1, also integrate CEO value into code and redo most things.. also write SQL query to update all existing firms, add ceo column and change type of CFO/coo column to string

**Review by thecsw — CHANGES_REQUESTED** [review]
Yeah, as we have discussed, changing the CEO, CFO, and COO entries into an array of strings will take more than just that. They are tightly integrated into the whole system, so for that you have to modify all the scripts in the codebase, where instead of incrementing values we concatenate strings or append values to the array.

Also, our current database runs the numerical values, so we will have to do some very careful SQL magic to add/edit/delete columns and we will have to make a script that will populate all of the values. That might be tricky.

On this note, this is a pretty heavy change and as we are already working on the new Go version of the bot, I would urge you to join the development of #394 

Please feel free to dispute any of the requests. If you have any question, I am always available on the common channels

**Comment by keanucz** [comment]
I am really not good at writing in Go but I can easily make this change in Python, will do.

**Comment by keanucz** [comment]
I've made a little SQL script that *should* migrate all the columns.. 
Bear with me, I haven't tested this so if you could test this in a similar DB environment then that would be great (I'm not good with SQL as you can see)

```sql
ALTER TABLE Firms ADD ceo VARCHAR(20) NOT NULL AFTER coo;
ALTER TABLE Firms MODIFY coo VARCHAR(20);
ALTER TABLE Firms MODIFY cfo VARCHAR(20);
UPDATE Firms INNER JOIN Investors ON Investors.firm = Firms.id AND Investors.firm_role = "ceo" SET Firms.ceo = Investors.name
UPDATE Firms INNER JOIN Investors ON Investors.firm = Firms.id AND Investors.firm_role = "coo" SET Firms.coo = Investors.name
UPDATE Firms INNER JOIN Investors ON Investors.firm = Firms.id AND Investors.firm_role = "cfo" SET Firms.cfo = Investors.name
```

**Comment by thecsw** [comment]
@Keanu73 There is a conflict with the master branch. 

Also, you can test the update with one of our database backups. See [ftp server](ftp://meme.market) and grab the `backup_01_28_2019.sql` to play with our data

our ftp is just ftp://meme.market or do `$ ftp meme.market`

**Comment by keanucz** [comment]
Alright thanks

**Comment by keanucz** [comment]
@thecsw I've figured out an SQL script that I have tested and does work, you can test it yourself:

```sql
ALTER TABLE Firms ADD ceo VARCHAR(20) NOT NULL AFTER coo;
ALTER TABLE Firms MODIFY coo VARCHAR(20);
ALTER TABLE Firms MODIFY cfo VARCHAR(20);

UPDATE firms 
LEFT JOIN investors ON firms.id = investors.firm
SET firms.ceo = investors.name 
WHERE investors.firm_role = "ceo";

UPDATE firms 
LEFT JOIN investors ON firms.id = investors.firm
SET firms.coo = investors.name 
WHERE investors.firm_role = "coo";

UPDATE firms 
LEFT JOIN investors ON firms.id = investors.firm
SET firms.cfo = investors.name 
WHERE investors.firm_role = "cfo"
```

**Comment by keanucz** [comment]
Do you want a screenshot of assurance that it works, or...??

**Comment by thecsw** [comment]
Approved! 
```
MariaDB [mysql]> select * from Firms where name = 'SOL Enterprises'
    -> ;
+-----+-----------------+-----------------+------+--------+----------------+----------------------+-------+--------+------+------+---------+-------------+
| id  | name            | balance         | size | coo    | ceo            | cfo                  | execs | assocs | tax  | rank | private | last_payout |
+-----+-----------------+-----------------+------+--------+----------------+----------------------+-------+--------+------+------+---------+-------------+
| 113 | SOL Enterprises | 170499018404036 |   41 | blkmmb | SlothySurprise | organic_crystal_meth |     7 |      9 |    5 |    4 |       1 |  1559340559 |
+-----+-----------------+-----------------+------+--------+----------------+----------------------+-------+--------+------+------+---------+-------------+
```

File: api/firms/firms.go [hunk 1]
```diff
@@ -36,7 +40,7 @@ func FirmsTop() func(w http.ResponseWriter, r *http.Request) {
 		defer conn.Close()
 		query := fmt.Sprintf(`
 SELECT id, name, balance, size, execs,
-tax, rank, private, last_payout
```
Comment (thecsw) [inline api/firms/firms.go:39]: Missing assocs here, I will add them

**Review by thecsw — APPROVED** [review]
Approved.

File: api/firms/firms.go (hunk 1 above)
Comment (keanucz) [reply api/firms/firms.go:39]: I think this was from when you accidentally undid the assocs API change I added, but it's alright

**Comment by keanucz** [comment]
Huzzah to API efficiency!

**Comment by keanucz** [comment]
I think I missed a couple of things.. e.g. https://github.com/Keanu73/memeinvestor_bot/blob/997fd0b0469e3c61198873258a2c1b53ba55443e/src/comment_worker.py#L630 and https://github.com/Keanu73/memeinvestor_bot/blob/997fd0b0469e3c61198873258a2c1b53ba55443e/src/comment_worker.py#L527

Since CEOs are able to leave firms or swap with someone else, I haven't implemented this so the changes wouldn't really go through to the ceo value.. could you throw in a quick fix maybe?

**Pull request merged**